python_modules/dagster/dagster/core/asset_defs/assets.py

```diff
@@ -61,3 +77,28 @@ def get_partition_mapping(self, in_asset_key: AssetKey) -> PartitionMapping:
             in_asset_key,
             self._partitions_def.get_default_partition_mapping(),
         )
+
+    def with_replaced_asset_keys(
+        self,
+        output_asset_key_replacements: Mapping[AssetKey, AssetKey],
+        input_asset_key_replacements: Mapping[AssetKey, AssetKey],
+    ) -> "AssetsDefinition":
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=ExperimentalWarning)
+
+            return self.__class__(
```

Comment:
these methods are always scary since its easy to introduce bugs where new optional fields on the class don't get added everywhere and then fail to get copied along. Having them in the same file like this is good. Maybe worth leaving comment notes under the last arg in the init signature to point towards these like [1]